tox.ini

```diff
@@ -6,9 +6,19 @@ extend-ignore =
     # See https://github.com/PyCQA/pycodestyle/issues/373
     E203,
 
+[tox]
+envlist = towncrier-check,docs-lint,pre-commit,py{38,39,310,311,312}
+skip_missing_interpreters = True
+
+[testenv:pre-commit]
+skip_install = True
+deps = ./core[dev]
+commands = pre-commit run --all-files --show-diff-on-failure --color=always
+
 # The leading comma generates the "py" environment.
 [testenv:py{,38,39,310,311,312}]
```

Comment:
note: these tox envs can't be run in parallel...or at least not reliably. the simultaneous `pip install` commands _might_ be an issue but I know that `coverage combine` is definitely an issue.